enterprise/server/util/chunker/chunker.go

```diff
@@ -72,19 +79,36 @@ func New(ctx context.Context, averageSize int, writeChunkFn WriteFunc) (*Chunker
 				return
 			}
 			if err != nil {
-				pr.CloseWithError(status.InternalErrorf("failed to get the next chunk: %s", err))
+				err = status.InternalErrorf("failed to get the next chunk: %s", err)
+				pr.CloseWithError(err)
+				c.mu.Lock()
+				defer c.mu.Unlock()
+				if c.err == nil {
+					c.err = err
+				}
 				return
 			}
 			if err := c.writeChunkFn(chunk.Data); err != nil {
-				pr.CloseWithError(status.InternalErrorf("writeChunkFn failed: %s", err))
+				err = status.InternalErrorf("writeChunkFn failed: %s", err)
+				pr.CloseWithError(err)
+				c.mu.Lock()
+				defer c.mu.Unlock()
+				if c.err == nil {
+					c.err = err
+				}
 				return
 			}
 		}
 	}()
 
 	go func() {
 		<-ctx.Done()
-		pw.CloseWithError(ctx.Err())
+		pr.CloseWithError(ctx.Err())
```

Comment:
move `pr.CloseWithError(ctx.Err())` below the Lock()

otherwise it's possible that the other goroutine will run first and update c.err before this one has a chance to